sematic/retry_settings.py

```diff
@@ -0,0 +1,41 @@
+# Standard Library
+from dataclasses import dataclass, field
+from typing import Tuple, Type
+
+# Sematic
+from sematic.utils.exceptions import ExceptionMetadata
+
+
+@dataclass
+class RetrySettings:
+    """
+    Configuration object to pass to `@sematic.func` to activate retries.
+
+    Parameters
+    ----------
+    exceptions: Tuple[Type[Exception]]
+        A tuple of exception types to retry for.
+
+    times: int
+        How may times to retry.
+    """
+
+    exceptions: Tuple[Type[Exception]]
```

Comment:
We might want to have the calculator call some `validate` method on this object when it is passed in. I can see it being a common mistake to do things like this:

```suggestion
    RetrySettings(
        exceptions=TimeoutError,  # not a tuple
        times=2,
    )
    
    RetrySettings(
        exceptions=(TimeoutError),  # still not a tuple
        times=2,
    )
```

Reply:
In general I try to stay away from "accepts whatever you pass to it" and instead be more strict. This limits the amount of validation code we need to write.
tuple vs set, most users do not no sets, and tuples are pretty common to pass list of classes.

I will add the validation that Josh suggested.